mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/dto/query/CriterionDTO.java

```diff
@@ -17,7 +17,7 @@
  * under the License.                                           *
  ****************************************************************/
 
-package org.apache.james.webadmin.vault.routes.query;
```

Comment:
Yeah, it's already causing me troubles... 
I will create 2 dto i think one for the event system, one for the webadmin.

Reply:
But i'm not sure it's a good thing either to duplicate all the dtos and serialization of the query part.

Reply:
The more i think about it, the more i think it would be a mistake to duplicate the code.
The operators,  fieldNames, combinators are enum. And the valids criterions are a limited subset of the combination of those.
By decoupling this it would be too easy to have valid query for the api that we will not be able to serialize/deserialize in the event store.

What do you think of keeping only one implementation shared between the two projects for the moment and adding a contract test on the web admin API for the query with a separate json file ? 


But for me the best solution would be to have the dto representing the query shared between the API and event system. With one JSON deserializer for the Event system and another for the API. This way wee keep a consistent representation of the query and we could add special/compatibility rule for the API if the need arise later.

Reply:
Thank you for thinking deeply about that. I trust you about the conclusion.